src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -197,13 +204,16 @@ protected void reloadConfig() {
     }
     try {
       log.info(String.format("Reloading hdfs-site.xml from %s", configUri));
-      String cfgCmd[] = new String[]{"sh", "-c",
-          String.format("curl -o hdfs-site.xml %s ; cp hdfs-site.xml etc/hadoop/", configUri)};
-      Process process = Runtime.getRuntime().exec(cfgCmd);
+      ProcessBuilder processBuilder = new ProcessBuilder("sh", "-c",
+          String.format("curl -o hdfs-site.xml %s && cp hdfs-site.xml etc/hadoop/", configUri));
+      Process process = processBuilder.start();
       //TODO(nicgrayson) check if the config has changed
       redirectProcess(process);
       int exitCode = process.waitFor();
       log.info("Finished reloading hdfs-site.xml, exited with status " + exitCode);
+      if (exitCode != 0) {
+        log.error("Error reloading hdfs-site.xml.");
```

Comment:
Minor: Could be an if/else with the log.info above, so you don't see "Finished reloading... Error reloading"